src/pages/sighting/identification/buildMatchingSetQuery.js

```diff
@@ -0,0 +1,58 @@
+import { flattenDeep, get } from 'lodash-es';
+
+function addChildMatches(rootRegion, matchId, insideRegion) {
+  const matchFound = rootRegion?.id === matchId;
+  const insideRegionNew = matchFound || insideRegion;
```

Comment:
I'm  a little confused by this variable name. If a match is found at the current.... level, let's call it?.... why would insideRegionNew be true?

In other words, if it matches at root, why there be a new insideRegion?

Reply:
so `addChildMatches` is called with `insideRegion` set to true if the matching region has already been found and `rootRegion` is a descendant of that region. if that's not true yet and a match is found, then `insideRegion` will be true at the next level for all of the current region's descendants. hence `insideRegionNew`. open to other names if you can think of something better

Reply:
That makes more sense now. I'm ok with that name... Also ok with something like currentlyInsideRegion.